phase1-coordinator/src/coordinator_state.rs

```diff
@@ -1994,38 +1990,54 @@ impl CoordinatorState {
                 let mut all_disposed_tasks: HashSet<Task> = participant_info.completed_tasks.iter().cloned().collect();
 
                 // A HashMap of tasks represented as (chunk ID, contribution ID) pairs.
-                let tasks_by_chunk: HashMap<u64, u64> = tasks.iter().map(|task| task.to_tuple()).collect();
+                let dropped_affected_tasks_by_chunk: HashMap<u64, u64> =
+                    dropped_affected_tasks.iter().map(|task| task.to_tuple()).collect();
 
                 // For every contributor we check if there are affected tasks. If the task
                 // is affected, it will be dropped and reassigned
                 for contributor_info in self.current_contributors.values_mut() {
+                    tracing::debug!("Checking contributor: {:#?}", &contributor_info);
+
                     // If the pending task is in the same chunk with the dropped task
                     // then it should be recomputed
                     let (disposing_tasks, pending_tasks) = contributor_info
                         .pending_tasks
                         .iter()
                         .cloned()
-                        .partition(|task| tasks_by_chunk.get(&task.chunk_id()).is_some());
+                        .partition(|task| dropped_affected_tasks_by_chunk.get(&task.chunk_id()).is_some());
 
                     // TODO: revisit the handling of disposing_tasks
                     //       https://github.com/AleoHQ/aleo-setup/issues/249
+
+                    // Technically this shouldn't be required, as the
+                    // disposing tasks will not be on disk yet because
+                    // they are still being computed by the contributor.
+                    // Probably doesn't hurt to keep this here anyway for
+                    // sake of consistency.
+                    affected_tasks.extend(&disposing_tasks);
```

Comment:
This change and line 2040 are the actual bugfix so far for this pull request, the rest is just refactoring, which can probably be moved to a separate pull request.